packages/connectivity/connectivity_macos/macos/Classes/ConnectivityPlugin.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import CoreWLAN
+import FlutterMacOS
+import Reachability
+import SystemConfiguration.CaptiveNetwork
+
+public class ConnectivityPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  var reach: Reachability?
+  var eventSink: FlutterEventSink?
+  var cwinterface: CWInterface?
```

Comment:
Can this be set immutably? It's a little cleaner to be able to do that than have to use null aware operators everywhere. I assume there may be some performance issue that means this needs to live in `init()`, but wanted to double check.

Reply:
The problem is that all these are set by methods executed later in the class lifecycle. Swift requires that immutable properties are set from the initialization of the class. In this case optionals are ok since force unwrapping is discouraged, and setting empty values at init would still require lots of checking

Reply:
Right, I was wondering if we could set ` cwinterface = CWWiFiClient.shared().interface()` immediately instead of in `init()`, specifically. I'm guessing there's a reason we want that call to be delayed though?